### PR TITLE
feat(action/patch coverage)

### PR DESCRIPTION
**Pull request by ymcorrea**
This PR should implement a patch / diff coverage feature in a Github action.

### CODE REVIEW

File: src/coverage.ts [hunk 1]
```diff
@@ -49,13 +66,18 @@ export const coverage = async ({ token, ghToken, summary }: Props) => {
   const comparison = res?.check
   const utoken = res?.metadata?.token
 
+  let patchCoverage: number | undefined
```
Comment (megheaiulian) [inline src/coverage.ts:69]: Move this into its own function that receives all the params it needs and returns the patch coverave  and some more meta information if needed.

File: src/coverage.ts [hunk 2]
```diff
@@ -49,13 +66,18 @@ export const coverage = async ({ token, ghToken, summary }: Props) => {
...
-  const diff = comparison
-    ? rate - comparison.linesHit / comparison.linesFound
-    : undefined
+  const diff = comparison ? rate - (comparison.linesHit / comparison.linesFound) : undefined
 
   core.debug(`rate: ${rate}`)
   core.debug(`diff: ${diff}`)
+  core.debug(`patchCoverage: ${patchCoverage}`)
 
   const message = ((): {
```
Comment (megheaiulian) [inline src/coverage.ts:82]: We need to separate this in functions.
Make a `coverageStatus` function that creates the main total coverage status message.
Make a  `patchStatus` function that create the patch coverage status message.

File: src/main.ts [hunk 3]
```diff
@@ -16,9 +16,17 @@ export async function run(): Promise<void> {
     const { summary } = await analyze(path)
 
     const rate = summary.lines.hit / summary.lines.found
+
```
Comment (megheaiulian) [inline src/main.ts:19]: Good idea to property set the output.
I think we need to set more outputs: `coverage` - the total coverage,  `diff` the diff coverage(between total coverages), `patch ` - the patch coverage.

File: src/coverage.ts [hunk 4]
```diff
@@ -81,5 +104,5 @@ export const coverage = async ({ token, ghToken, summary }: Props) => {
     ...message
   })
 
-  return { status, rate, diff }
+  return { status, rate, diff, patchCoverage }
```
Comment (megheaiulian) [inline src/coverage.ts:107]: We return a bunch of stuff here but we mainly need coverage, diff & patch.

**Review by megheaiulian — CHANGES_REQUESTED** [review]
Good work.

Left a couple of comments.

We should probably add a test where we mock the data returned by the github api or plug our own diff file and then check the patch coverage in a test.
We also run the action on the repo itself and should see the patch coverrage result even in this PR I think ...

File: src/coverage.ts (hunk 1 above)
Comment (ymcorrea) [reply src/coverage.ts]: done

File: src/coverage.ts (hunk 2 above)
Comment (ymcorrea) [reply src/coverage.ts]: done

File: src/coverage.ts (hunk 4 above)
Comment (ymcorrea) [reply src/coverage.ts]: done

**Comment by megheaiulian** [comment]
@ymcorrea There are some conflicts...

**Pull request closed without merging**